tests/plugins/test_hide_files.py

```diff
@@ -0,0 +1,244 @@
+                RUN mv -f /etc/yum.repos.d/repo_ignore_1.repo /tmp || :
+                RUN mv -f /etc/yum.repos.d/repo_ignore_2.repo /tmp || :
+                USER {0}
+                RUN yum install -y python-flask
+                CMD /bin/bash
+                USER 0
+                RUN mv -fZ /tmp/repo_ignore_1.repo /etc/yum.repos.d/repo_ignore_1.repo || :
+                RUN mv -fZ /tmp/repo_ignore_2.repo /etc/yum.repos.d/repo_ignore_2.repo || :
+                USER {0}
+                """.format(inherited_user))
+        elif user_in_df:
+            expected_df_content = dedent("""\
+                FROM sha256:123456
+                RUN mv -f /etc/yum.repos.d/repo_ignore_1.repo /tmp || :
+                RUN mv -f /etc/yum.repos.d/repo_ignore_2.repo /tmp || :
+                RUN yum install -y python-flask
+                USER {0}
+                CMD /bin/bash
+                USER 0
+                RUN mv -fZ /tmp/repo_ignore_1.repo /etc/yum.repos.d/repo_ignore_1.repo || :
+                RUN mv -fZ /tmp/repo_ignore_2.repo /etc/yum.repos.d/repo_ignore_2.repo || :
+                USER {0}
+                """.format(user_in_df))
+        else:
+            expected_df_content = dedent("""\
+                FROM sha256:123456
+                RUN mv -f /etc/yum.repos.d/repo_ignore_1.repo /tmp || :
+                RUN mv -f /etc/yum.repos.d/repo_ignore_2.repo /tmp || :
+                RUN yum install -y python-flask
+                CMD /bin/bash
+                RUN mv -fZ /tmp/repo_ignore_1.repo /etc/yum.repos.d/repo_ignore_1.repo || :
+                RUN mv -fZ /tmp/repo_ignore_2.repo /etc/yum.repos.d/repo_ignore_2.repo || :
+                """)
+
+        assert df.content == expected_df_content
+
+    def test_hide_files_multi_stage(self, tmpdir):
+        df_content = dedent("""\
+            FROM sha256:123456 as builder
+            RUN blah
+            USER custom_user
+            RUN bluh
+
+            FROM sha256:654321
+            RUN yum install -y python-flask
+            USER custom_user2
+            CMD /bin/bash
+            """)
+        df = df_parser(str(tmpdir))
+        df.content = df_content
+        hide_files = {'tmpdir': '/tmp', 'files': ['/etc/yum.repos.d/repo_ignore_1.repo',
+                                                  '/etc/yum.repos.d/repo_ignore_2.repo']}
+        parent_images = {
+            ImageName.parse('fedora'): ImageName.parse('sha256:123456'),
+            ImageName.parse('python'): ImageName.parse('sha256:654321'),
+        }
+
+        tasker, workflow = self.prepare(
+            df.dockerfile_path, hide_files=hide_files, parent_images=parent_images,
+            inherited_user="inherited_user")
+
+        runner = PreBuildPluginsRunner(tasker, workflow, [
+            {'name': HideFilesPlugin.key, 'args': {}},
+        ])
+        runner.run()
+
+        expected_df_content = dedent("""\
+            FROM sha256:123456 as builder
+            USER 0
```

Comment:
Just spotted that inject_yum_repo uses `USER root` -- should we follow that in this plugin too?
https://github.com/projectatomic/atomic-reactor/blob/master/atomic_reactor/plugins/pre_inject_yum_repo.py#L47

Reply:
done